app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoFragment.java

```diff
@@ -117,16 +115,9 @@ public final void onClick(final View v) {
             case R.id.start_video_button:
                 // Start thread reading on server
                 try {
-                    // TODO(ksheremet): check errors
-                    final SharedPreferences sharedPreferences =
-                            PreferenceManager.getDefaultSharedPreferences(getActivity());
-                    final String host = sharedPreferences
-                            .getString(Settings.VIDEO_HOST.toString(), "");
-                    final int port = Integer.parseInt(sharedPreferences
-                            .getString(Settings.VIDEO_PORT.toString(), ""));
-                    final String password = sharedPreferences
-                            .getString(Settings.VIDEO_PASSWORD.toString(), "");
-                    mVideoThread = new Thread(new VideoDecoderRunnable(host, port, password));
+                    SharedPreferencesHandler handler = new SharedPreferencesHandler(getActivity());
+                    mVideoThread = new Thread(new VideoDecoderRunnable(handler.getVideoHost(),
```

Comment:
Since you're touching this, as Codacy noticed `mVideoThread` can be just a local variable.